### PR TITLE
docs: Add DataFusion subprojects to navigation menu, other minor updates

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

Add links to DataFusion subproject to make them more easily discoverable

## What changes are included in this PR?

- Added links to menu
- Moved bdt from the list of active projects to the list of less active projects
- Reordered list of projects so that they are in alphabet order

## Are these changes tested?

Tested locally

## Are there any user-facing changes?

No

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -116,3 +116,13 @@ Please see the `developer’s guide`_ for contributing and `communication`_ for
...
    contributor-guide/governance
    contributor-guide/specification/index
+
+.. _toc.contributor-guide:
+
+.. toctree::
+   :maxdepth: 1
+   :caption: DataFusion Subprojects
+
+   DataFusion Ballista <https://arrow.apache.org/ballista/>
```
Comment (andygrove) [inline docs/source/index.rst:126]: Ballista has not yet published its site to the new location

**Review by alamb — APPROVED** [review]
Looks good to me. Thank you @andygrove 

Note I rendered the docs and the subproject links are so far down I had to scroll down:

Screenshot 2024-05-03 at 7 08 48 AM

Maybe we should reduce the maxdepth for the library guide

**Review by viirya — APPROVED** [review]

**Comment by andygrove** [comment]
Maybe we could move the ASF Links section to the end?

**Comment by alamb** [comment]
> Maybe we could move the ASF Links section to the end?

Maybe we could shrink them too (there is no reason for each to get its own vertical line)

Like we could collapse the License / Security into one line maybe

Likewise Donate/Thanks 🤔

**Review by comphead — APPROVED** [review]
lgtm thanks @andygrove

**Pull request merged**